datafusion/common/src/file_type.rs

```diff
@@ -281,18 +311,18 @@ impl FileType {
             FileType::JSON | FileType::CSV => Ok(format!("{}{}", ext, c.get_ext())),
             FileType::PARQUET | FileType::AVRO | FileType::ARROW => match c.variant {
                 UNCOMPRESSED => Ok(ext),
-                _ => internal_err!(
-                    "FileCompressionType can be specified for CSV/JSON FileType."
-                ),
+                _ => Err(DataFusionError::Internal(
```

Comment:
I tried to go back over this file and use the macro, but ran into some issues. Took a while to figure out you can't import it within the same crate, but have to specify it like:

`crate::error::_not_impl_err!`

In places where `?` is used in combination with the macro e.g. 

`crate::error::_not_impl_err!("message")?`

I was getting a cannot convert into DataFusionError compiler message. 🤔

Reply:
Actually, figured this out. I was trying to substitute in the macro in a place it shouldn't go (where DataFusionError is needed, not Err(DataFusionError)).